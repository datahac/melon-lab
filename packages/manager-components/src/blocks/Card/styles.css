.card {
  background-color: var(--color-white);
  cursor: pointer;
  display: table-row;
}

.card--active {
  background-color: var(--color-secondary);
}

.card:hover {
  background-color: color-mod(var(--color-secondary) a(50%));
}

.card__info {
  display: flex;
  color: var(--color-grey);
  font-size: var(--font-size-s);
  line-height: 1;
  padding-top: var(--space-xxs);
  justify-content: space-between;

  @media (--viewport-m) {
    padding: 0;
    width: 50%;
  }
}

.card__rank {
  padding: 0 var(--space-m) 0 var(--space-m);
  font-size: var(--font-size-xl);
  display: table-cell;
  vertical-align: middle;
  border-bottom: 1px solid var(--color-border);
}

.card__name {
  font-weight: var(--font-weight-bold);
  line-height: 1;

  @media (--viewport-m) {
    flex-grow: 1;
  }
}

.card__inception-date {
  flex-grow: 1;
  text-align: right;
}

.card__info-cell {
  width: 100%;
  padding: var(--space-s) var(--space-m) var(--space-s) 0;
  display: table-cell;
  border-bottom: 1px solid var(--color-border);
  vertical-align: middle;
}

.card__info-wrap {
  flex-grow: 1;
  align-items: center;

  @media (--viewport-m) {
    display: flex;
  }
}

.card__label {
  font-weight: var(--font-weight-bold);
  display: block;
  padding-bottom: var(--space-xxs);
}
