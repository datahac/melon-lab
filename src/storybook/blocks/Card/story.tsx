import { storiesOf } from '@storybook/react';
import React from 'react';
import Card from './index';

const data = {
  name: 'Black Turtle',
  address: '0x5bBA9263Ab1eA26FF9c0FeE3619e7AAf7C79E02b',
  inception: '26. Jul 2018 10:48',
  sharePrice: {
    base: {
      quantity: '1',
      token: {
        symbol: 'ETH',
        decimals: 4,
      },
    },
    quote: {
      quantity: '1',
      token: {
        symbol: 'ETH',
        decimals: 4,
      },
    },
  },
  rank: 1,
};

storiesOf('Blocks|Card', module).add('Default', () => {
  return <Card {...data} />;
});
