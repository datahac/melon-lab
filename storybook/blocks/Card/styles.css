.card {
  background-color: var(--color-white);
  cursor: pointer;
  display: table-row;
}

.card:hover,
.card--active {
  background-color: var(--color-primary);
}

.card__info {
  display: flex;
  color: var(--color-grey);
  font-size: var(--font-size-s);
  line-height: 1;
  padding-top: var(--space-xxs);
  justify-content: space-between;
  flex-grow: 1;

  @media (--viewport-m) {
    padding: 0;
    width: 50%;
    flex-grow: unset;
  }
}

.card__rank {
  padding: 0 var(--space-m) 0 var(--space-m);
  font-size: var(--font-size-xl);
  display: table-cell;
  vertical-align: middle;
  border-bottom: 1px solid var(--color-border);
}

.card__name {
  font-weight: var(--font-weight-bold);
  line-height: 1;
  width: 100%;

  @media (--viewport-m) {
    flex-grow: 1;
    margin-right: var(--space-m);
    width: auto;
  }
}

.card__gav, .card__share-price, .card__inception-date, .card__version {
  margin: 0 40px;
}

.card__info-cell {
  width: 100%;
  padding: var(--space-s) var(--space-m) var(--space-s) 0;
  display: table-cell;
  border-bottom: 1px solid var(--color-border);
  vertical-align: middle;
}

.card__info-wrap {
  flex-grow: 1;
  align-items: center;

  display: flex;
  flex-wrap: wrap;
}

.card__label {
  font-weight: var(--font-weight-bold);
  display: block;
  padding-bottom: var(--space-xxs);
}

.card__rank-symbol {
  font-size: var(--font-size-xs);
}
