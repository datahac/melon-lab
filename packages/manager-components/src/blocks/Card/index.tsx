import classNames from 'classnames';
import React, { StatelessComponent } from 'react';

import styles from './styles.css';

export interface CardProps {
  address?: string;
  inception?: string;
  isActive?: boolean;
  name?: string;
  onClick?: (id) => void;
  rank?: number;
  sharePrice?: string;
}

const Card: StatelessComponent<CardProps> = ({
  address,
  inception,
  isActive,
  name,
  onClick,
  rank,
  sharePrice,
}) => {
  const cardClassNames = classNames('card', {
    'card--active': isActive,
  });

  const onCardClick = () => onClick && onClick(address);

  return (
    <div onClick={onCardClick} className={cardClassNames}>
      <style jsx>{styles}</style>
      <div className="card__rank">
        <span className="card__rank-symbol">#</span>
        {rank}
      </div>
      <div className="card__info-cell">
        <div className="card__info-wrap">
          <div className="card__name">{name}</div>
          <div className="card__info">
            <div className="card__share-price">
              <span className="card__label">Share price</span> {sharePrice}
            </div>
            <div className="card__inception-date">
              <span className="card__label">Inception Date</span> {inception}
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Card;
