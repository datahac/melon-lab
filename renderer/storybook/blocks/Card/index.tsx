import classNames from 'classnames';
import React, { StatelessComponent } from 'react';
import format from 'date-fns/format';
import displayPrice from '~/shared/utils/displayPrice';
import Link from '~/blocks/Link';
import Icons from '~/design/Icons';
import * as Tm from '@melonproject/token-math';

import styles from './styles.css';
import displayQuantity from '~/shared/utils/displayQuantity';

export interface CardProps {
  inception?: string;
  isActive?: boolean;
  name?: string;
  rank?: number;
  sharePrice?: Tm.PriceInterface;
  onClick?: React.MouseEventHandler;
  decimals?: number;
}

const Card: StatelessComponent<CardProps> = ({
  inception,
  isActive,
  name,
  gav,
  rank,
  version,
  sharePrice,
  onClick,
  decimals = 4,
}) => {
  const cardClassNames = classNames('card', {
    'card--active': isActive,
  });

  return (
    <div onClick={onClick} className={cardClassNames}>
      <style jsx>{styles}</style>
      <div className="card__rank">
        <span className="card__rank-symbol">#</span>
        {rank}
      </div>
      <div className="card__info-cell">
        <div className="card__info-wrap">
          <div className="card__name">{name}</div>
          <div className="card__info">
            <div className="card__share-price">
              <span className="card__label">Share price</span>{' '}
              {sharePrice && displayPrice(sharePrice, decimals)}
            </div>
            <div className="card__aum">
              <span className="card__label">AUM</span>{' '}
              {gav && displayQuantity(gav, decimals)}
            </div>
            <div className="card__inception-date">
              <span className="card__label">Inception Date</span>{' '}
              {format(inception, 'DD. MMM YYYY HH:mm')}
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Card;
