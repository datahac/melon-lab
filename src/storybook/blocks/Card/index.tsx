import classNames from 'classnames';
import React, { StatelessComponent } from 'react';
import format from 'date-fns/format';
import displayPrice from '~/utils/displayPrice';
import Link from '~/blocks/Link';
import Icon from '~/blocks/Icon';

import styles from './styles.css';

export interface CardProps {
  inception?: string;
  isActive?: boolean;
  name?: string;
  rank?: number;
  sharePrice?: string;
  reportUrl?: string;
  onClick?: React.MouseEventHandler;
  decimals?: number;
}

const Card: StatelessComponent<CardProps> = ({
  inception,
  isActive,
  name,
  rank,
  sharePrice,
  reportUrl,
  onClick,
  decimals = 4,
}) => {
  const cardClassNames = classNames('card', {
    'card--active': isActive,
  });

  const handleReportClick = e => e.stopPropagation();

  return (
    <div onClick={onClick} className={cardClassNames}>
      <style jsx>{styles}</style>
      <div className="card__rank">
        <span className="card__rank-symbol">#</span>
        {rank}
      </div>
      <div className="card__info-cell">
        <div className="card__info-wrap">
          <div className="card__name">{name}</div>
          <div className="card__info">
            <div className="card__share-price">
              <span className="card__label">Share price</span>{' '}
              {sharePrice && displayPrice(sharePrice, decimals)}
            </div>
            <div className="card__inception-date">
              <span className="card__label">Inception Date</span>{' '}
              {format(inception, 'DD. MMM YYYY HH:mm')}
            </div>
          </div>
          <div className="card__report" onClick={handleReportClick}>
            <Link target="_blank" href={reportUrl} title="Show Report">
              <Icon width="16px" height="20px" name="icons_report" />
            </Link>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Card;
