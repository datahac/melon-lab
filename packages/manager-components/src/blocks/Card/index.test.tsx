import React from 'react';
import Card from './index';

describe('Card', () => {
  const defaultElement = <Card />;

  it('should render correctly', () => {
    const wrapper = shallow(defaultElement);
    expect(wrapper).toMatchSnapshot();
  });
});
