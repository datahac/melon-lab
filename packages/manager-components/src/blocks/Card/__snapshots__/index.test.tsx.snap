// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render correctly 1`] = `
<div
  className="jsx-__scopedHash card"
  onClick={[Function]}
>
  <JSXStyle
    css="__scoped"
    styleId="__scopedHash"
  />
  <div
    className="jsx-__scopedHash card__rank"
  >
    #
  </div>
  <div
    className="jsx-__scopedHash card__info-cell"
  >
    <div
      className="jsx-__scopedHash card__info-wrap"
    >
      <div
        className="jsx-__scopedHash card__name"
      />
      <div
        className="jsx-__scopedHash card__info"
      >
        <div
          className="jsx-__scopedHash card__share-price"
        >
          <span
            className="jsx-__scopedHash card__label"
          >
            Share price:
          </span>
           
        </div>
        <div
          className="jsx-__scopedHash card__inception-date"
        >
          <span
            className="jsx-__scopedHash card__label"
          >
            Inception Date:
          </span>
           
        </div>
      </div>
    </div>
  </div>
</div>
`;
